legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/core/FunctionExecutionSupport.java

```diff
@@ -124,6 +121,45 @@ default String getExecutionKey(PackageableElement element, Map<String, Object> a
         return "";
     }
 
+    static Iterable<? extends LegendExecutionResult> execute(AbstractLSPGrammarExtension extension, SectionState section, String entityPath, String commandId, Map<String, String> executableArgs, Map<String, Object> inputParameters)
+    {
+        switch (commandId)
+        {
+            case FunctionExecutionSupport.EXECUTE_QUERY_ID:
+            {
+                return FunctionExecutionSupport.executeQuery(extension, section, entityPath, executableArgs, inputParameters);
+            }
+            case FunctionExecutionSupport.GENERATE_EXECUTION_PLAN_ID:
+            {
+                return FunctionExecutionSupport.generateExecutionPlan(extension, section, entityPath, executableArgs, inputParameters);
+            }
+            case FunctionExecutionSupport.GRAMMAR_TO_JSON_LAMBDA_ID:
+            {
+                return FunctionExecutionSupport.convertGrammarToLambdaJson(extension, section, entityPath, executableArgs, inputParameters);
+            }
+            case FunctionExecutionSupport.JSON_TO_GRAMMAR_LAMBDA_BATCH_ID:
+            {
+                return FunctionExecutionSupport.convertLambdaJsonToGrammarBatch(extension, section, entityPath, executableArgs, inputParameters);
+            }
+            case FunctionExecutionSupport.GET_LAMBDA_RETURN_TYPE_ID:
+            {
+                return FunctionExecutionSupport.getLambdaReturnType(extension, section, entityPath, executableArgs, inputParameters);
+            }
+            case FunctionExecutionSupport.SURVEY_DATASETS_ID:
+            {
+                return FunctionExecutionSupport.generateDatasetSpecifications(extension, section, entityPath, executableArgs, inputParameters);
+            }
+            case FunctionExecutionSupport.CHECK_DATASET_ENTITLEMENTS_ID:
+            {
+                return FunctionExecutionSupport.generateEntitlementReports(extension, section, entityPath, executableArgs, inputParameters);
+            }
+            default:
+            {
+                return extension.defaultExecute(section, entityPath, commandId, executableArgs, inputParameters);
```

Comment:
Let's return null, and the caller then make the decision what to do.  Or, we can just throw with command not supported...

Reply:
Done (throwing an error now)